### PR TITLE
GH1051 Drop astype tests for Mac arm for float128 and complex256

### LINKED ISSUE #1051: astype tests failing for `f16`, `float128`, `c32` and `complex256` on MacOS arm64 platforms
**Describe the bug**

MacOS arm64 (M1 and above) platforms do not recognize some dtypes from 

```
FAILED tests/test_series.py::test_astype_float['f16'-<class 'numpy.longdouble'>] - TypeError: data type 'f16' not understood
FAILED tests/test_series.py::test_astype_float['float128'-<class 'numpy.longdouble'>] - TypeError: data type 'float128' not understood
FAILED tests/test_series.py::test_astype_complex['c32'-<class 'numpy.clongdouble'>] - TypeError: data type 'c32' not understood
FAILED tests/test_series.py::test_astype_complex['complex256'-<class 'numpy.clongdouble'>] - TypeError: data type 'complex256' not understood
```

**To Reproduce**
Run `poetry run poe test_all` on a Mac M1 or above (for reference I ran a build on GitHub Actions, see https://github.com/loicdiridollou/pandas-stubs/actions/runs/11981847576/job/33408829929)

**Please complete the following information:**
 - OS: MacOS
 - OS Version 15
 - python version 3.10/3.11/3.12/3.13
 - version of type checker NA
 - version of installed `pandas-stubs` NA

### PR DESCRIPTION
**Pull request by loicdiridollou**
- [x] Closes #1051
- [x] Tests added: Please use `assert_type()` to assert the type of any return value

### CODE REVIEW

**Comment by loicdiridollou** [comment]
Needed to let poetry choose the right version of `tables`, wondering if it is okay, at least we will need that to move to supporting python3.13.

File: pyproject.toml [hunk 1]
```diff
@@ -47,7 +47,7 @@ black = ">=23.3.0"
 isort = ">=5.12.0"
 openpyxl = ">=3.0.10"
 # for tables, MacOS gives random CI failures on 3.9.2
-tables = { version = "==3.9.2", python = "<4" }   # 3.8.0 depends on blosc2 which caps python to <4
+tables = { version = ">=3.9.2", python = "<4" }   # 3.8.0 depends on blosc2 which caps python to <4
```
Comment (Dr-Irv) [inline pyproject.toml:50]: version 3.10.1 is out, and it was what was used in the tests, so changing this from 3.9.2 to 3.10.1 is warranted.

**Review by Dr-Irv — CHANGES_REQUESTED** [review]
I think we should update the version of `tables`

**Review by Dr-Irv — APPROVED** [review]
thanks @loicdiridollou

**Pull request merged**